lib/validators/one_favorite_per_owner_validator.rb

```diff
@@ -0,0 +1,16 @@
+module Validators
+  # Validator for Collection model
+  # Validate that only one favorite collection is created per owner for each project
+  class OneFavoritePerOwnerValidator < ActiveModel::Validator
+    def validate(new_record)
+      return unless new_record.owner
+
+      records = new_record.owner.collections
+      return unless records && new_record.favorite
```

Comment:
will records be a AR:Relation here? i.e. `[]` but we don't test for empty?

also the guard clause on `new_record.favorite` should be at the top not here, short circuit and avoid any db work etc if we can.

Reply:
Ah yes, very true. Moving the guard clause 👍